File: app/src/main/java/za/org/grassroot2/services/account/GrassrootAuthService.java
package za.org.grassroot2.services.account;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

import javax.inject.Inject;

import timber.log.Timber;
import za.org.grassroot2.GrassrootApplication;
import za.org.grassroot2.services.rest.GrassrootAuthApi;

public class GrassrootAuthService extends Service {

    @Override
    public IBinder onBind(Intent intent) {
        AccountAuthenticator authenticator = new AccountAuthenticator(this);
        Timber.e("bound the auth service");
        return authenticator.getIBinder();
    }

}

File: app/src/main/java/za/org/grassroot2/services/account/GrassrootAuthService.kt
package za.org.grassroot2.services.account

import android.app.Service
import android.content.Intent
import android.os.IBinder

class GrassrootAuthService : Service() {

//    override fun onBind(intent: Intent): IBinder? {
//        val authenticator = AccountAuthenticator(this)
//        Timber.e("bound the auth service")
//        return authenticator.iBinder
//    }

    override fun onBind(intent: Intent): IBinder? = AccountAuthenticator(this).iBinder

}
